blockchain/chain.go

```diff
@@ -1039,7 +1044,12 @@ func (b *BlockChain) reorganizeChain(detachNodes, attachNodes *list.List) error
 		// to it.  Also, provide an stxo slice so the spent txout
 		// details are generated.
 		stxos := make([]SpentTxOut, 0, countSpentOutputs(block))
-		err = view.connectTransactions(block, &stxos)
+		if n.parent != nil && b.IsMagneticAnomalyEnabled(n.parent.hash) {
```

Comment:
After activation this check can/will change to simply:
```go
if height > b.chainparmas.MagneticAnomalyActivationHeight 
```
So this squirrely logic isn't permanent.